src/mcpwm.rs

```diff
@@ -0,0 +1,597 @@
+//! Motor Control Pulse Width Modulator peripheral
```

Comment:
Random thought, what is the reasoning with [esp-rs/esp-hal](https://github.com/esp-rs/esp-hal) is there any sort of whish if keeping this sort of compatible with that where possible? If so is there anything we want to try and keep in mind here that would have to be done differently if this were to be added in esp-hal? Or that would be made differently when not having the IDF in between.

Reply:
I wouldn't worry about the esp-hal implementation for now, that would be completely different. The only thing we _may_ want to think about is keeping the same API through some MCPWM traits, but its hard to get  this sort of thing right on the first try. Let us use this PR to experiment and get an initial implementation, we can refine and refactor with follow up PRs.